### DISCUSSION TITLE
Displaying Invalid ID on search bar

### DISCUSSION THREAD

**Discussion by cbjones93** [nss-day-cohort-47/C47-Discussions#58] [Help Me] [answered]
Working on displaying an alert message for an invalid Id, however my code will produce an unending loop that will crash the browser. I'm looking for something that will either run the alert once & refresh the page or alert once and clear the search box.
Is there a way to use .then with this scenario?
```navElement.addEventListener('keypress', event => {
	if (event.key === 'Enter' && event.target.id === "mySearch") {
		const legoValue = document.querySelector("input[name='searchBox']").value
		console.log(`user wants to see `, legoValue)
		showFilteredSearch(legoValue);
	}
})

const showFilteredSearch = (searchValue) => {
	const filteredSearch = useLegos().filter(singleLego => {
		if (singleLego.LegoId === searchValue) {
			return singleLego
		} else if (singleLego.LegoId !== searchValue) {
			return alert("not valid id")
		}
	})
	makeLegoList(filteredSearch);
}
```

here's what I have googled, although they didn't seem very helpful but maybe I just can't see it.
https://stackoverflow.com/questions/24768067/display-alert-only-once
https://javascript.info/alert-prompt-confirm
https://www.javatpoint.com/javascript-alert

**Comment by Chadwell-Clark** [comment, marked as answer]
I had a similar situation trying to use an alert. I was getting an alert when I refreshed the page mainly because I was getting an undefined value. 
I also do not think that you can use a .then() as most of your code above is not going to be asynchronous. 
Bryan helped me with setting the value of the search to an empty string which causes the default placeholder to be displayed in the search input area when you are done with a search, or after a failed search. 
He also helped me walk through what we are looking for to trigger an error, which would be an empty array when doing the search.

**Reply by cbjones93** [reply]
This worked! Thank you so much!

**Comment by BrysonMG** [comment]
I had a similar issue where the alert kept popping up over and over. What fixed it for me was taking it out of the for loop it was in and calling it after the loop finished; however, your's doesn't seem to be inside a loop that I can see. This was my solution:

I declared an empty array that took in all the blocks that "passed" the filter, then I passed that array into the function that starts the chain of events that get it to display on the DOM. After I called that function, I made an if statement that checks if the array is empty, and if it is, displays the alert.
